Automattic/wp-calypso
# Domain Management i1: refactored table proposal based on React.Context

Related to p1693555174915309/1693475126.737279-slack-C04JYPF9AV6

## Proposed Changes

* Move all `DomainTable` state into a hook that uses `React.Context` to provide it to children.
* External API remains the same. See `packages/domains-table/src/domains-table/index.tsx`, which forwards `props` onto the internal components. 

Tables can now be composed like so. Child components, for example, `<DomainsTableToolbar/>` can easily be omitted or replaced by a custom one. 

```ts
import { DomainsTable as InternalDomainsTable, DomainsTableProps } from './domains-table';
import { DomainsTableBody } from './domains-table-body';
import { DomainsTableHeader } from './domains-table-header';
import { DomainsTableToolbar } from './domains-table-toolbar';

import './style.scss';

export function DomainsTable( props: DomainsTableProps ) {
	return (
		<InternalDomainsTable { ...props }>
			<DomainsTableToolbar />

			<table>
				<DomainsTableHeader />
				<DomainsTableBody />
			</table>

		</InternalDomainsTable>
	);
}
```


## Testing Instructions



* On this branch, visit `/domains/manage?flags=domains/management` and verify the table works as before. 
* Try removing `<DomainsTableToolbar/>` from `<Example/>` or replacing it with something else. 
* Create your own `<Example/>` that includes a `<tfoot/>` component. 

## Pre-merge Checklist
